### PR TITLE
New update: Implementation for SIR Model

### PR DESCRIPTION
**Pull request by kle6951**
I have implemented the SIR model #12  that follows the coding style used by Bryce and Tobias. Please review the code and let me know if I can do anything to improve it and fix any errors if needed.

### CODE REVIEW

**Comment by bryce-carson** [comment]
Thanks, @kle6951. I am beginning the review now.

A quick question, does your other pull request (#32) depend on this one? It appears you modified (or created) SIR_solve.R in that pull request. The names of the two PRs indicate they should be, or may be, isolated. That's why I ask about the dependency issue; I will review this PR first, since it is older and there appears to be some dependency.

**Comment by kle6951** [comment]
@bryce-carson no, I don’t think I have created any pull request associated with #32. I have no clue why it shows up like that. Thank you for pointing this out.

**Comment by bryce-carson** [comment]
> @bryce-carson no, I don’t think I have created any pull request associated with #32. I have no clue why it shows up like that. Thank you for pointing this out.

Khanh, #32 _is_ your other pull-request. Issues and pull requests are numbered in the same way (a number/pound sign/symbol and numbered serially and jointly). If a new repository had two issues created, with a pull request created after the first issue was made but before the second issue was made, they would be respectively numbered one, two, and three.

**Comment by kle6951** [comment]
Is there anything you need me to do to fix it or you can take a look at it?

File: R/SIR_solve.R [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+{
+  ggplot2::ggplot(solveSIR(), ggplot2::aes(x = time)) +
+    plotTheme +
+    ggplot2::labs(title = "SIRS Epidemic Model", y = "Number of People",
+                  x = "Time") +
+    ggplot2::scale_x_continuous(expand = c(0, 0)) +
+    ggplot2::scale_y_continuous(expand = c(0, 0)) +
+    ggplot2::geom_line(aes(y = S, color = "Blue"), linewidth = 1.5) +
+    ggplot2::geom_line(aes(y = I, color = "Red"), linewidth = 1.5) +
+    ggplot2::geom_line(aes(y = R, color = "Green"), linewidth = 1.5) +
```
Comment (bryce-carson) [inline R/SIR_solve.R:57]: Please include `ggplot2::` when calling `aes()` here.

File: R/SIR_solve.R [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      name = "SIR", breaks = c("Blue", "Red", "Green"),
+      labels = c("Susceptible", "Infected", "Recovered"), guide = "legend"
+    )
+}
+
+# plot phase plane
+plotPhasePlaneSIR <- function()
+{
+  ggplot2::ggplot(solveSIR(), ggplot2::aes(x = S))+
+    ggplot2::geom_line(aes(y = I, color = "Blue"), linewidth = 1.5) +
```
Comment (bryce-carson) [inline R/SIR_solve.R:68]: Include `ggplot2::` when calling `aes()` here.

File: R/SIR_solve.R [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    plotTheme +
+    ggplot2::ggtitle("SI Phase Plane") + 
+    ggplot2::theme(plot.title = element_text(size = 22, face = "bold")) +
```
Comment (bryce-carson) [inline R/SIR_solve.R:71]: Include `ggplot2::` when calling `element_text()` here.

**Review by bryce-carson — CHANGES_REQUESTED** [review]
There are only a few changes that need to happen here before I merge this PR. Thanks, @kle6951.

**Comment by bryce-carson** [comment]
@kle6951, as you work in your editor, you may find it helpful to resolve each code review comment as you fix them so you know you resolve them all.

**Comment by kle6951** [comment]
@bryce-carson I have done the changes as requested. Please let me know again if anything to be fixed. Just want to inform you that I also updated the code for the SIRS and LaTex on the weekend in case you haven't seen them. Please check it out.

**Review by bryce-carson — APPROVED** [review]

**Pull request merged**